lib/utils.js

```diff
@@ -146,3 +150,57 @@ utils.registerSourceMap = () => {
 utils.registerTypeScript = () => {
   require('ts-node').register();
 };
+
+/**
+ * Check if given template is installed and return details about it
+ * 
+ * @private
+ * @param {String} name name of the template
+ * @param {String} PACKAGE_JSON_FILENAME standard name of the package.json file
+ * @returns {Object|undefined} object with fields like pkgPath, name, version
+ */
+utils.getTemplateDetails = (name, PACKAGE_JSON_FILENAME) => {
+  let pkgPath;
+  let pkgPackageJsonPath;
+  let installedPkg;
+  
+  //first trying to resolve package by its path
+  //this could be done at the end, without additional if but the advantage if below approach is that we explicitly know resolving was successful by the template file path and we can log proper debug info
+  if (utils.isFileSystemPath(name)) {
+    pkgPath = path.resolve(name);
+    
+    log.debug('Remember that your local template must have its own node_modules installed first, "npm install" is not triggered by the generator.');
+  } else {
+    //first trying to resolve package in local dependencies of generator or the project that uses generator library
+    //return path doesn't point to package.json
+    pkgPath = resolvePkg(name);
+    //resolvePkg returns undefined if dependency is not found in local dependencies
+
+    //no we need to look for the template in global location
+    //we use resolveFrom with package.json reference as normal resolve fails on template because they do not have entrypoint ("main" property) as packages normally do.
+    if (!pkgPath) {
+      try {
+        //trying to find in global yarn location
+        pkgPackageJsonPath = resolveFrom(globalDirs.yarn.packages, path.join(name, PACKAGE_JSON_FILENAME));
+      } catch (_) {
+        //trying to find in global npm location
+        pkgPackageJsonPath = resolveFrom(globalDirs.npm.packages, path.join(name, PACKAGE_JSON_FILENAME));
+      }
+
+      if (pkgPackageJsonPath) {
+        log.debug(`${name} not found in local dependencies but found it installed as a global package.`);
+        //removing package.json from path
+        pkgPath = pkgPackageJsonPath.substring(0, pkgPackageJsonPath.lastIndexOf('/'));
```

Comment:
for cross platform support it's better to use `path.dirname(pkgPackageJsonPath
)`.

Reply:
love it mate, thanks! fixed